enhancements/windows-containers/disconnected-support.md

```diff
@@ -0,0 +1,372 @@
+workloads easily and successfully in disconnected clusters. This is an extremely important ask for customer environments
+where Windows nodes must pull images from air-gapped registries for security reasons.
+The added benefit here is that the changes proposed in this enhancement are not restricted to disconnected environments;
+users in __connected__ clusters can take advantage of this mirror registry support. This would allow better performance
+as images can be stored in registries closer to the production environment, reducing pull times for notoriously large 
+Windows container images, in addition the security benefits of restricted networks.
+
+### Goals
+
+* Create a mechanism for WMCO to consume global mirror registry settings from existing platform resources (IDMS/ITMSes),
+  such that it enables Windows nodes to pull images in disconnected environments
+* Configure the mirror registry settings in the containerd runtime, a WMCO-managed component on Windows nodes 
+* React to changes to the mirror registry settings during WMCO runtime
+* Maintain normal functionality in connected clusters and clusters without any cluster-wide mirror registry configuration
+
+### Non-Goals
+
+* First-class support of [all OpenShift image registry setting configurations](https://docs.openshift.com/container-platform/4.14/openshift_images/image-configuration.html#images-configuration-file_image-configuration), including the `Image` resource
+* Supporting clusters that use only `ImageContentSourcePolicy` resources. These were [deprecated in 4.13](https://docs.openshift.com/container-platform/4.14/release_notes/ocp-4-14-release-notes.html#ocp-4-14-deprecated-removed-features) 
+and users [should migrate to IDMSes](https://docs.openshift.com/container-platform/4.14/openshift_images/image-configuration.html#images-configuration-registry-mirror-convert_image-configuration) instead.
+* Using [`ImageContentPolicy` resources](https://docs.openshift.com/container-platform/4.14/rest_api/config_apis/imagecontentpolicy-config-openshift-io-v1.html)
+
+
+## Proposal
+
+### Design Details
+
+WMCO will use the mirrors data in `ImageDigestMirrorSet` and `ImageTagMirrorSet` resources 
+to create a set of [hosts.toml files](https://github.com/containerd/containerd/blob/main/docs/hosts.md#hoststoml-content-description---detail),
+one for each registry mirror, that will be transferred to the Windows instance.
+This will be done on during initial node configuration, similarly to how bootstrap files (like the kubelet config file)
+are generated and transferred. This way nodes will have the right mirror registry config right when they join the cluster.
+
+In addition, the operator should also react to any events on IDMS/ITMS resources through a new "registry controller" --
+when mirror settings are updated, WMCO will then react by:
+- generating an updated set of hosts.toml files
+- copying over the new registry config files to each Windows instance, if a file has changed
```

Comment:
This is a pattern we should be careful with. WMCO should avoid SSHing into Windows nodes to react to cluster changes. Think about this with a large amount of nodes, and frequent changes to the mirrors. Maintaining the state of dynamic configuration files is something we should make use of the windows-services configmap to do.